pkg/agent/multicast/vif_allocator.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	multicastsyscall "antrea.io/antrea/third_party/unix"
+
+	"k8s.io/klog/v2"
+)
+
+// VIF stands for virtual interface for multicast in kernel multicast implementation
+// and is used for adding/deleting static multicast entries, please check
+// https://github.com/torvalds/linux/blob/master/include/uapi/linux/mroute.h#L80
+// VIFAllocator is used for saving vif <-> interface mapping.
```

Comment:
VIFAllocator is used for allocating VIF and caching VIF <-> interface mapping.

Reply:
updated